mobile/lib/shared/providers/asset.provider.dart

```diff
@@ -231,6 +237,24 @@ class AssetNotifier extends StateNotifier<AssetsState> {
 
     return newAsset.isFavorite;
   }
+
+  Future<bool> toggleArchive(Asset asset, bool status) async {
+    final newAsset = await _assetService.changeArchiveStatus(asset, status);
+
+    if (newAsset == null) {
+      log.severe("Change favorite status failed for asset ${asset.id}");
+      return asset.isArchived;
+    }
+
+    final index = state.allAssets.indexWhere((a) => asset.id == a.id);
+    if (index > 0) {
+      state.allAssets[index] = newAsset;
+      _updateAssetsState(state.allAssets);
+      getAllAsset();
```

Comment:
@fyfrey `_updateAssetsState` doesn't update the database, so when I go to the archive page, the assets list isn't updated. So I put `getAllAsset` in here, I don't think it is the optimized solution, I did try to write to database but it doesn't help either. Probably due to the way I read the asset list in https://github.com/immich-app/immich/pull/2258/files#r1167682409. Any suggestion from your side

Reply:
yeah.. `getAllAsset` is really bad there.
I've modified the AssetService to update assets also in the local DB after changes on the server were successful.
Further, the state is now updated by removing archived assets, only updating their `isArchived` flag has no effect as there are already in the list of assets to be rendered